src/NHapi.Base/Util/MessageIterator.cs

```diff
@@ -1,584 +1,707 @@
+            {
+                return Name.GetHashCode() + (700 * Rep);
+            }
+
+            /// <summary>
+            /// Override to string.
+            /// </summary>
+            /// <returns></returns>
+            public override string ToString()
+            {
+                return Name + ":" + Rep;
+            }
+        }
+
+        /// <summary> A structure position within a message. </summary>
+        public class Position
+        {
+            /// <summary>
+            /// The parent.
+            /// </summary>
+            [Obsolete("This method has been replaced by 'Parent'.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.NamingRules",
+                "SA1307:Accessible fields should begin with upper-case letter",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.MaintainabilityRules",
+                "SA1401:Fields should be private",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            public IGroup parent;
+
+            /// <summary>
+            /// The index.
+            /// </summary>
+            [Obsolete("This method has been replaced by 'Index'.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.NamingRules",
+                "SA1307:Accessible fields should begin with upper-case letter",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage(
+                "StyleCop.CSharp.MaintainabilityRules",
+                "SA1401:Fields should be private",
+                Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+            public Index index;
+
+            /// <summary>
+            /// The position of the element.
+            /// </summary>
+            /// <param name="parent">Parent.</param>
+            /// <param name="name">Name.</param>
+            /// <param name="rep">Repetition.</param>
+            public Position(IGroup parent, string name, int rep)
+            {
+                Parent = parent;
+                Index = new Index(name, rep);
+            }
+
+            /// <summary>
+            /// The position of the element.
+            /// </summary>
+            /// <param name="parent">Parent.</param>
+            /// <param name="i">index.</param>
+            public Position(IGroup parent, Index i)
+            {
+                Parent = parent;
+                Index = i;
+            }
+
+            /// <summary>
+            /// Gets or sets the parent.
+            /// </summary>
+            public IGroup Parent

```

Comment:
Are fields needed here? convert to standard props and get fields?
same below

Reply:
the fields are public so i've added the props and marked the fields as obsolete